Add TS examples for getting-started pages of the Auth category

devnotfound opened the pull request:
#### Description of changes:
Add TS examples for getting-started pages of the Auth category
#### Related GitHub issue #, if available:

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [X] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [X] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

cwomack commented on src/fragments/lib/auth/js/emailpassword.mdx:
````diff
@@ -2,6 +2,43 @@
+```ts
+import { Auth } from 'aws-amplify';
+
+interface SignUpParams {
````
@devnotfound, would you mind changing the name of the interface here to TSSignUpParams to preserve the name "SignUpParams" and ensure there's no duplicate declarations of it?

cwomack commented on src/fragments/lib/auth/js/emailpassword.mdx:
````diff
@@ -39,6 +78,29 @@ The `Auth.signUp` promise returns a data object of type [`ISignUpResult`](https:
+```ts
+import { Auth } from 'aws-amplify';
+
+interface ResendConfCodeParams {
````
Same thing here, can we update the usage of `interface` to be a `type` instead?

cwomack commented on src/fragments/lib/auth/js/emailpassword.mdx:
````diff
@@ -51,10 +113,34 @@ async function resendConfirmationCode() {
+```ts
+import { Auth } from 'aws-amplify';
+
+interface ConfirmSignUpParams {
````
Same thing here, can we update the usage of `interface` to be a `type` instead?

cwomack commented on src/fragments/lib/auth/js/emailpassword.mdx:
````diff
@@ -67,13 +153,39 @@ async function confirmSignUp() {
+```ts
+import { Auth } from 'aws-amplify';
+
+interface ConfirmSignUpParams {
````
Same thing here, can we update the usage of `interface` to be a `type` instead?

cwomack commented on src/fragments/lib/auth/js/emailpassword.mdx:
```diff
@@ -51,10 +113,34 @@ async function resendConfirmationCode() {
+  code: string;
+}
+
+export async function confirmSignUp(confirmSignUpParams: ConfirmSignUpParams) {
```
Can you update this to include passing in `username` and `code` as a destructured object such as below:

`confirmSignUp({username, code}:ConfirmSignUpParams)`

cwomack commented on src/fragments/lib/auth/js/emailpassword.mdx:
```diff
@@ -2,6 +2,43 @@
+  phoneNumber: string;
+}
+
+export async function signUp(signUpParams: SignUpParams) {
```
Can you update this to include passing in `username`, `password`, `email`,  and `phoneNumber` as a destructured object such as below:

`signUp({username, password, email, phoneNumber}:SignUpParameters)`

cwomack commented on src/fragments/lib/auth/js/emailpassword.mdx:
```diff
@@ -129,7 +243,29 @@ Auth.signUp({
+  password: string;
+}
+
+export async function signIn(signInParams: SignInParams) {
```
Can you update this to include passing in `username` and `password` as a destructured object such as below:

`signIn({username, password}:signInParams)`

cwomack commented on src/fragments/lib/auth/js/emailpassword.mdx:
````diff
@@ -129,7 +243,29 @@ Auth.signUp({
+```ts
+import { Auth } from 'aws-amplify';
+
+interface SignInParams {
````
Same thing here, can we update the usage of `interface` to be a `type` instead?

cwomack commented:
@devnotfound, greatly appreciate you creating this PR! Left some comments for some updates and feel free to reply back with any questions.

devnotfound commented:
Hey @cwomack @abdallahshaban557 
I have addressed all the feedback, please let me know if I have missed anything. 
note: I have renamed all the types in the above examples that had the `Params` suffix to `Parameters`

cwomack reviewed (approved):
LGTM!

cwomack commented:
Appreciate the updates and quick turnaround @devnotfound!

israx reviewed (approved):
LGTM. Thanks for putting this together!

abdallahshaban557 reviewed (approved):
LGTM
